File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int result = 0;
        
        int r = 0;
        int small = prices[0];
        
        while (r<n){
            if (prices[r] < small){
                small = prices[r];
            } else if (prices[r] - small > result){
                result = prices[r] - small;
            }
            r++;
        }
        return result;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        cur_max = prices[0]
        glob_max = 0
        for i in range(1,len(prices)):
            price = prices[i]
            cur_max = min(price, cur_max)
            glob_max = max(glob_max, price-cur_max)
        return glob_max